lsp-completion.el

```diff
@@ -495,19 +495,20 @@ Others: TRIGGER-CHARS"
 
     (cond
      ((equal lsp-completion-provider :none))
-     ((and (null lsp-completion-provider)
-           (functionp 'company-lsp))
-      (progn
-        (company-mode 1)
-        (add-to-list 'company-backends 'company-lsp)
-        (setq-local company-backends (remove 'company-capf company-backends))))
+
+     ((and lsp-completion-provider
```

Comment:
This will pass even with `lsp-completion-provider` equal to `:capf` and `(member 'company-lsp company-backends)`.
Shouldn't you combine two checks into one and check showing the warning when `(null lsp-completion-provider)`?